cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -1425,6 +1544,19 @@ data TxValidityUpperBound era where
 deriving instance Eq   (TxValidityUpperBound era)
 deriving instance Show (TxValidityUpperBound era)
 
+instance Semigroup (TxValidityUpperBound era) where
+    TxValidityNoUpperBound _ <> x = x
+    x <> TxValidityNoUpperBound _ = x
+    TxValidityUpperBound lWit l <> TxValidityUpperBound rWit r = TxValidityUpperBound (lWit <> rWit) (min l r)
+
+instance IsCardanoEra era => Monoid (TxValidityUpperBound era) where
+    mempty = case cardanoEra @era of
+      ByronEra -> TxValidityNoUpperBound ValidityNoUpperBoundInByronEra
+      ShelleyEra -> TxValidityUpperBound ValidityUpperBoundInShelleyEra maxBound
```

Comment:
```
ShelleyEra -> TxValidityUpperBound ValidityUpperBoundInShelleyEra maxBound
```

Was this how it was in the original code?

Why is Shelly era special that it should use `maxBound`?

Reply:
```
-- Note that the 'ShelleyEra' /does not support/ omitting a validity upper
-- bound. It was introduced as a /required/ field in Shelley and then made
-- optional in Allegra and subsequent eras.
--
-- The Byron era supports this by virtue of the fact that it does not support
-- validity ranges at all.
```